app/models/container.rb

```diff
@@ -37,4 +39,14 @@ def parent_id=(val)
   def set_project_to_parent_project
     self.project = self.parent.project if self.parent
   end
+
+  def set_deleted_from_parent
```

Comment:
I think it would be good to break this method into two methods. Something like `delete_to_trash` and `restore_to_parent`. They could encapsulate setting `is_deleted` as well as `deleted_from_parent_id` and `parent_id`. This should result in code that is easier to both read and test. 🤞